src/schemathesis/specs/openapi/expressions/lexer.py

```diff
@@ -0,0 +1,104 @@
+from enum import Enum, unique
+from typing import Callable, Generator
+
+import attr
+
+
+@unique  # pragma: no mutate
+class TokenType(Enum):
+    VARIABLE = 1  # pragma: no mutate
+    STRING = 2  # pragma: no mutate
+    POINTER = 3  # pragma: no mutate
+    DOT = 4  # pragma: no mutate
+    LBRACKET = 5  # pragma: no mutate
+    RBRACKET = 6  # pragma: no mutate
+
+
+@attr.s(slots=True)  # pragma: no mutate
+class LexToken:
+    value: str = attr.ib()  # pragma: no mutate
+    type_: TokenType = attr.ib()  # pragma: no mutate
+
+    @classmethod
+    def variable(cls, value: str) -> "LexToken":
+        return cls(value, TokenType.VARIABLE)
+
+    @classmethod
+    def string(cls, value: str) -> "LexToken":
+        return cls(value, TokenType.STRING)
+
+    @classmethod
+    def pointer(cls, value: str) -> "LexToken":
+        return cls(value, TokenType.POINTER)
+
+    @classmethod
+    def lbracket(cls) -> "LexToken":
+        return cls("{", TokenType.LBRACKET)
+
+    @classmethod
+    def rbracket(cls) -> "LexToken":
+        return cls("}", TokenType.RBRACKET)
+
+    @classmethod
+    def dot(cls) -> "LexToken":
+        return cls(".", TokenType.DOT)
+
+    @property
+    def is_string(self) -> bool:
+        return self.type_ == TokenType.STRING
+
+    @property
+    def is_variable(self) -> bool:
+        return self.type_ == TokenType.VARIABLE
+
+    @property
+    def is_dot(self) -> bool:
+        return self.type_ == TokenType.DOT
+
+
+def tokenize(expression: str) -> Generator[LexToken, None, None]:
```

Comment:
Probably tokenization result could be cached to avoid re-tokenizing the same expressions